File: app/src/main/java/com/androidevlinux/percy/UTXO/data/network/apis/BitstampApiImpl.java
package com.androidevlinux.percy.UTXO.data.network.apis;

import com.androidevlinux.percy.UTXO.data.models.bitstamp.BitstampBean;
import com.androidevlinux.percy.UTXO.utils.NativeUtils;

import io.reactivex.Observable;

/**
 * Created by percy on 5/12/17.
 */

public class BitstampApiImpl extends AbstractBaseApi<BitstampAPI> {

    private static BitstampApiImpl bitstampApiManager;
    private BitstampAPI bitstampAPI;
    private BitstampApiImpl(){
        setBaseUrl(NativeUtils.getbitStampBaseUrl());
        bitstampAPI = getClient(BitstampAPI.class);

    }

    public static BitstampApiImpl getInstance(){
        if(bitstampApiManager==null)
            bitstampApiManager = new BitstampApiImpl();
        return bitstampApiManager;
    }

    public Observable<BitstampBean> getBitstampTicker(String symbol){
        return bitstampAPI.getBitstampTicker(symbol);
    }
}

File: app/src/main/java/com/androidevlinux/percy/UTXO/data/network/apis/BitstampApiImpl.kt
package com.androidevlinux.percy.UTXO.data.network.apis

import com.androidevlinux.percy.UTXO.data.models.bitstamp.BitstampBean
import com.androidevlinux.percy.UTXO.utils.NativeUtils

import io.reactivex.Observable

/**
 * Created by percy on 5/12/17.
 */

class BitstampApiImpl private constructor() : AbstractBaseApi<BitstampAPI>() {
    private val bitstampAPI: BitstampAPI

    init {
        setBaseUrl(NativeUtils.bitStampBaseUrl)
        bitstampAPI = getClient(BitstampAPI::class.java)

    }

    fun getBitstampTicker(symbol: String): Observable<BitstampBean> {
        return bitstampAPI.getBitstampTicker(symbol)
    }

    companion object {

        private var bitstampApiManager: BitstampApiImpl? = null

        val instance: BitstampApiImpl
            get() {
                if (bitstampApiManager == null)
                    bitstampApiManager = BitstampApiImpl()
                return bitstampApiManager!!
            }
    }
}
